events/janus_rabbitmqevh.c

```diff
@@ -0,0 +1,724 @@
+			janus_flags_reset(&janus_rabbitmqevh.events_mask);
+		} else if(!strcasecmp(item->value, "all")) {
+			/* Subscribe to everything */
+			janus_flags_set(&janus_rabbitmqevh.events_mask, JANUS_EVENT_TYPE_ALL);
+		} else {
+			/* Check what we need to subscribe to */
+			gchar **subscribe = g_strsplit(item->value, ",", -1);
+			if(subscribe != NULL) {
+				gchar *index = subscribe[0];
+				if(index != NULL) {
+					int i=0;
+					while(index != NULL) {
+						while(isspace(*index))
+							index++;
+						if(strlen(index)) {
+							if(!strcasecmp(index, "sessions")) {
+								janus_flags_set(&janus_rabbitmqevh.events_mask, JANUS_EVENT_TYPE_SESSION);
+							} else if(!strcasecmp(index, "handles")) {
+								janus_flags_set(&janus_rabbitmqevh.events_mask, JANUS_EVENT_TYPE_HANDLE);
+							} else if(!strcasecmp(index, "jsep")) {
+								janus_flags_set(&janus_rabbitmqevh.events_mask, JANUS_EVENT_TYPE_JSEP);
+							} else if(!strcasecmp(index, "webrtc")) {
+								janus_flags_set(&janus_rabbitmqevh.events_mask, JANUS_EVENT_TYPE_WEBRTC);
+							} else if(!strcasecmp(index, "media")) {
+								janus_flags_set(&janus_rabbitmqevh.events_mask, JANUS_EVENT_TYPE_MEDIA);
+							} else if(!strcasecmp(index, "plugins")) {
+								janus_flags_set(&janus_rabbitmqevh.events_mask, JANUS_EVENT_TYPE_PLUGIN);
+							} else if(!strcasecmp(index, "transports")) {
+								janus_flags_set(&janus_rabbitmqevh.events_mask, JANUS_EVENT_TYPE_TRANSPORT);
+							} else if(!strcasecmp(index, "core")) {
+								janus_flags_set(&janus_rabbitmqevh.events_mask, JANUS_EVENT_TYPE_CORE);
+							} else {
+								JANUS_LOG(LOG_WARN, "Unknown event type '%s'\n", index);
+							}
+						}
+						i++;
+						index = subscribe[i];
+					}
+				}
+				g_strfreev(subscribe);
+			}
+		}
+	}
+
+	/* Handle configuration, starting from the server details */
+	item = janus_config_get_item_drilldown(config, "general", "host");
+	if(item && item->value)
+		rmqhost = g_strdup(item->value);
+	else
+		rmqhost = g_strdup("localhost");
+	int rmqport = AMQP_PROTOCOL_PORT;
+	item = janus_config_get_item_drilldown(config, "general", "port");
+	if(item && item->value)
+		rmqport = atoi(item->value);
+
+	/* Credentials and Virtual Host */
+	item = janus_config_get_item_drilldown(config, "general", "vhost");
+	if(item && item->value)
+		vhost = g_strdup(item->value);
+	else
+	vhost = g_strdup("/");
+	item = janus_config_get_item_drilldown(config, "general", "username");
+	if(item && item->value)
+		username = g_strdup(item->value);
+	else
+		username = g_strdup("guest");
+	item = janus_config_get_item_drilldown(config, "general", "password");
+	if(item && item->value)
+		password = g_strdup(item->value);
+	else
+		password = g_strdup("guest");
+
+	/* SSL config*/
+	item = janus_config_get_item_drilldown(config, "general", "ssl_enable");
+	if(!item || !item->value || !janus_is_true(item->value)) {
+		JANUS_LOG(LOG_INFO, "RabbitMQ SSL support disabled\n");
```

Comment:
Long lines should clarify that we're referring to the RabbitMQ *event handler*, otherwise this might be confused with long lines from the RabbitMQ transport instead. Spotted this but there may be others.